tests/validator_test.go

```diff
@@ -320,13 +324,415 @@ var _ = Describe("Template validator", func() {
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Should have given error failing to fulfill validations")
+			Expect(apiClient.Get(ctx, client.ObjectKey{Name: vmi.Name, Namespace: vmi.Namespace}, vmi)).ToNot(HaveOccurred())
+			WaitForVMIPhase(vmi, v1.Running, 180)
+			time.Sleep(time.Second * 100)
+			logs, err := GetVMIPodLogs(vmi)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(logs).To(ContainSubstring("warning.*Memory size not within range:"))
+		})
+		It("test with partial annotations", func() {
+			vmi.Spec.Domain.Machine.Type = "q35"
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse("128M"),
+			}
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores: 2,
+			}
+			vm = test_utils.NewRandomVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template-namespace": strategy.GetNamespace(),
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("Test vm with UI style annotations", func() {
+			vmi.Spec.Domain.Machine.Type = "q35"
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse("128M"),
+			}
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores: 2,
+			}
+			vm = test_utils.NewRandomVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           templateWithRules.Name,
+				"vm.kubevirt.io/template.namespace": templateWithRules.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("Test vm with template info in labels", func() {
+			vmi.Spec.Domain.Machine.Type = "q35"
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse("128M"),
+			}
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores: 2,
+			}
+			vm = test_utils.NewRandomVirtualMachine(vmi, false)
+			vm.ObjectMeta.Labels = map[string]string{
+				"vm.kubevirt.io/template":           templateWithRules.Name,
+				"vm.kubevirt.io/template.namespace": templateWithRules.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("test template with incomplete CPU info", func() {
+			vmi.Spec.Domain.Machine.Type = "q35"
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse("128M"),
+			}
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Sockets: 1,
+			}
+			vm = test_utils.NewRandomVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           templateWithRules.Name,
+				"vm.kubevirt.io/template-namespace": templateWithRules.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("[test_id:5173]: should create a VM that passes validation", func() {
```

Comment:
Please create a new `Context` to group all the tests for validation rules inside the VM's annotations.